packages/core/crates/core-network/src/ping.rs

```diff
@@ -34,43 +37,60 @@ use utils_misc::time::wasm::current_timestamp;
 
 const PINGS_MAX_PARALLEL: usize = 14;
 
-#[cfg_attr(test, mockall::automock)]
-pub trait PingExternalAPI {
-    fn on_finished_ping(&self, peer: &PeerId, result: crate::types::Result);
-}
-
 /// Basic type used for internally aggregating ping results to be further processed in the
 /// PingExternalAPI callbacks.
 type PingMeasurement = (PeerId, crate::types::Result);
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct PingConfig {
-    pub max_parallel_pings: usize,
-    pub environment_id: String,
-    pub normalized_version: String,
-    pub timeout: Duration,
+    max_parallel_pings: usize,
+    environment_id: String,
+    normalized_version: String,
+    timeout: Duration,
 }
 
-#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
-pub struct Ping {
+impl PingConfig {
+    pub fn new(
+        max_parallel_pings: usize,
+        environment_id: String,
+        normalized_version: String,
+        timeout: Duration,
+    ) -> Self {
+        Self {
+            max_parallel_pings,
+            environment_id,
+            normalized_version,
+            timeout,
+        }
+    }
+}
```

Comment:
Given no real construction limitation and relative independence of the config options, this `impl` block could be removed to leave only the in-place struct initialization, i.e. this reads well in the code:

```rust
let ping_config = PingConfig {
    max_parallel_pings: 10,
    environment_id: "monte_rosa_3000".to_owned(),
    normalized_version: "2.7.3000".to_owned(),
    timeout: std::time::Duration::from_secs(2),
}
```